internal/sql/migrations.go

```diff
@@ -428,6 +430,77 @@ var (
 					return performMigration(ctx, tx, migrationsFs, dbIdentifier, "00029_autopilot_2", log)
 				},
 			},
+			{
+				ID: "00030_remove_contract_sets",
+				Migrate: func(tx Tx) error {
+					// prepare statement to rename the buffer
+					stmt, err := tx.Prepare(ctx, "UPDATE slab_buffers SET filename = ? WHERE filename = ?")
+					if err != nil {
+						return fmt.Errorf("failed to prepare update statement: %w", err)
+					}
+					defer stmt.Close()
+
+					// prepare a helper to safely copy and sync the file
+					copyBuffer := func(buffer string) (string, error) {
+						parts := strings.Split(buffer, "-")
+						if len(parts) != 4 {
+							return "", fmt.Errorf("invalid buffer filename '%s'", buffer)
+						}
+
+						src, err := os.Open(buffer)
+						if err != nil {
+							return "", fmt.Errorf("failed to open buffer: %w", err)
+						}
+						defer src.Close()
+
+						dstFile, err := os.Create(strings.Join(parts[1:], "-"))
+						if err != nil {
+							return "", fmt.Errorf("failed to create destination buffer: %w", err)
+						}
+						defer dstFile.Close()
+
+						_, err = io.Copy(dstFile, src)
+						if err != nil {
+							return "", fmt.Errorf("failed to copy buffer: %w", err)
+						}
+
+						err = dstFile.Sync()
+						if err != nil {
+							return "", fmt.Errorf("failed to sync buffer: %w", err)
+						}
+
+						return dstFile.Name(), nil
+					}
+
+					// fetch all buffers
+					buffers, err := m.SlabBuffers(ctx, tx)
+					if err != nil {
+						return err
+					}
+
+					// copy all buffers
+					for _, buffer := range buffers {
+						if dst, err := copyBuffer(buffer); err != nil {
+							return fmt.Errorf("failed to copy buffer '%s': %w", buffer, err)
+						} else if res, err := stmt.Exec(ctx, buffer, dst); err != nil {
+							return fmt.Errorf("failed to update buffer '%s': %w", buffer, err)
+						} else if n, err := res.RowsAffected(); err != nil {
+							return fmt.Errorf("failed to fetch rows affected: %w", err)
+						} else if n != 1 {
+							return fmt.Errorf("failed to update buffer '%s': %w", buffer, err)
+						}
+					}
+					// remove original buffers
```

Comment:
Need to update the database to the new buffers first before deleting them from disk